distmod() for calculating the distance modulus

SzymonPrajs opened the pull request:
I've added a short function for calculating the distance modulus for a given cosmology and redshift. I use it all the time in Astropy so it might be quite useful here too.

kbarbary commented:
I felt like this was missing as well; good addition.

What's the need for `abs`? Is luminosity distance ever negative?

Can you add a test?

SzymonPrajs commented:
I wasn't sure if to include the  `abs`. I can't see how luminosity distance could ever be negative but I checked this against Astropy and they had it there so I that's why it was there originally. I removed it since it's almost certainly not necessary.
  
I've also added a test. Values are checked against iCosmos and and Astropy.

kbarbary commented:
Great!
